### PR TITLE
handle http context in xfcc authenticator

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn PTAL

**Comment by ramaraochavali** [comment]
@howardjohn added unit tests. PTAL

**Review by howardjohn — APPROVED** [review]

File: pkg/security/security.go [hunk 1]
```diff
@@ -367,6 +367,28 @@ type AuthContext struct {
...
+}
+
+// Header returns the authenticated remote address from AuthContext.
+func (ac *AuthContext) Header(header string) []string {
+	if ac.GrpcContext != nil {
+		if meta, ok := metadata.FromIncomingContext(ac.GrpcContext); ok {
+			return meta.Get(header)
+		}
+	} else if ac.Request != nil {
+		return ac.Request.Header[header]
```
[the remark this answers is not in the document]
Comment (ramaraochavali) [reply pkg/security/security.go:387]: I was wondering about that.  But I could not find an easy way to do that. Any suggestions?

[the remark this answers is not in the document]
Comment (ramaraochavali) [reply pkg/security/security.go:387]: Let me try that

**Pull request merged**